js/map/mapObject.mjs

```diff
@@ -169,6 +169,24 @@ MapIcon.prototype.draw = function(ctx, mouseLoc, currentSelection){
             ctx.drawImage(icons.Check, screenSpaceIconOrigin.x, screenSpaceIconOrigin.y, this.width(), this.height());
         }
     }
+
+    //Draw extra gate icons
+    if(this.cell.hive.classname == "location" && this.cell.name.includes("Oblivion Gate")){
+        let n = this.cell.name;
+        if(n.includes("Random") && randomGateCount >= 40){
+            ctx.drawImage(icons.Check, screenSpaceIconOrigin.x, screenSpaceIconOrigin.y, this.width(), this.height());
+        }
+        if(n.includes("Fixed")){
+            ctx.drawImage(icons.Fixed, screenSpaceIconOrigin.x, screenSpaceIconOrigin.y, this.width(), this.height());
+        }
+        if(n.includes("No Reroll")){    
+            ctx.drawImage(icons.No_Reroll, screenSpaceIconOrigin.x, screenSpaceIconOrigin.y, this.width(), this.height());
+        }
+        if(n.includes("2 Fame")){
```

Comment:
thanks i hate it

Reply:
idk of a better way at the moment though

Reply:
We could relabel some things. and then when we draw the icons it could be something like:
```
notes.split(", ").foreach(note => {
    ctx.drawImage(icons["overlay_" + note], *Etc...*);
});
```

I'm going to try and implement this, but leave it in it's own commit so we can change it if we want.